test/lib/test-helper.ts

```diff
@@ -6,59 +6,101 @@ import Promise = require('bluebird');
+                                      }
+                                     );
+        server.isHTTPS = https;
 
-    server.stdout.on('data', (data: any): void => {
-        log(data.toString());
-    });
-    server.stderr.on('data', (data: any): void => {
-        log(data.toString());
-    });
-    server.on('close', (): void => {
-        log('child process exited');
-    });
+        server.stdout.on('data', (data: any): void => {
+            log(data.toString());
+        });
+        server.stderr.on('data', (data: any): void => {
+            log(data.toString());
+        });
+        server.on('close', (): void => {
+            log('child process exited');
+        });
 
-    return (new Promise<ReturnValue>((resolve: (result: ReturnValue) => void,
-                                      reject: (error: Error) => void): void => {
-        server.on('message', (m: any): void => {
-            // Connect the ipc socket once test server is ready
-            var socket: any = io.connect('http://localhost:' + testport);
-            socket.emitAcknowledge = (event: string, arg?: any): Promise<void> => {
-                return (new Promise<void>((resolve: () => void,
-                                           reject: (error: Error) => void): void => {
-                    var argument = (2 === arguments.length) ? arg : null;
-                    socket.emit(event, argument, (): void => {
-                        resolve();
+        return (new Promise<ReturnValue>((resolve: (result: ReturnValue) => void,
+                                          reject: (error: Error) => void): void => {
+            server.on('message', (m: any): void => {
+                // Connect the ipc socket once test server is ready
+                var socket: any = io.connect('http://localhost:' + testport);
+                socket.emitAcknowledge = (event: string, arg?: any): Promise<void> => {
+                    return (new Promise<void>((resolve: () => void,
+                                               reject: (error: Error) => void): void => {
+                        var argument = (2 === arguments.length) ? arg : null;
+                        socket.emit(event, argument, (): void => {
+                            resolve();
+                        });
+                    }));
+                };
+                socket.on('connect', (): void => {
+                    resolve({
+                        'server': server,
+                        'ipc': socket
                     });
-                }));
-            };
-            socket.on('connect', (): void => {
-                resolve({
-                    'server': server,
-                    'ipc': socket
                 });
             });
-        });
-    }));
+        }));
+    });
+}
+
+export function stopServer(server: IOurChildProcess): Promise<void>
+{
+    var p: Promise<any> = server.isHTTPS ? removeCertificate() : Promise.resolve();
```

Comment:
There may be a more generic way to deal with clean-up that doesn't require `if`/`switch` type behavior that executes a queued up run-list or array of callbacks to execute for cleanup.